e2e/cluster/cmx/cluster.go

```diff
@@ -0,0 +1,463 @@
+	}
+
+	in.T.Logf("copying scripts to node %s", node.Name)
+	if err := copyScriptsToNode(node); err != nil {
+		return nil, fmt.Errorf("failed to copy scripts to node %s: %v", node.Name, err)
+	}
+
+	if index == 0 {
+		in.T.Logf("exposing port 30003 on node %s", node.Name)
+		hostname, err := exposePort(node, "30003")
+		if err != nil {
+			return nil, fmt.Errorf("failed to expose port: %v", err)
+		}
+		node.adminConsoleURL = fmt.Sprintf("http://%s", hostname)
+	}
+
+	return &node, nil
+}
+
+func ensureAssetsDir(node Node) error {
+	stdout, stderr, err := runCommandOnNode(node, []string{"mkdir", "-p", "/assets"})
+	if err != nil {
+		return fmt.Errorf("failed to create directory: %v: %s: %s", err, stdout, stderr)
+	}
+	return nil
+}
+
+func copyScriptsToNode(node Node) error {
+	// Create a temporary directory for the archive
+	tempDir, err := os.MkdirTemp("", "scripts-archive")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Create the archive
+	archivePath := filepath.Join(tempDir, "scripts.tgz")
+	output, err := exec.Command("tar", "-czf", archivePath, "-C", "scripts", ".").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to create scripts archive: %v: %s", err, string(output))
+	}
+
+	// Copy the archive to the node
+	if err := copyFileToNode(node, archivePath, "/tmp/scripts.tgz"); err != nil {
+		return fmt.Errorf("failed to copy scripts archive to node: %v", err)
+	}
+
+	// Extract the archive in /usr/local/bin
+	_, stderr, err := runCommandOnNode(node, []string{"tar", "-xzf", "/tmp/scripts.tgz", "-C", "/usr/local/bin"})
+	if err != nil {
+		return fmt.Errorf("failed to extract scripts archive: %v: %s", err, stderr)
+	}
+
+	// Clean up the archive on the node
+	_, stderr, err = runCommandOnNode(node, []string{"rm", "-f", "/tmp/scripts.tgz"})
+	if err != nil {
+		return fmt.Errorf("failed to clean up scripts archive: %v: %s", err, stderr)
+	}
+
+	return nil
+}
+
+func getSSHEndpoint(nodeID string) (string, error) {
+	cmd := exec.Command("replicated", "vm", "ssh-endpoint", nodeID)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to get SSH endpoint for node %s: %v: %s", nodeID, err, string(output))
+	}
+	return strings.TrimSpace(string(output)), nil
+}
+
+func (c *Cluster) Airgap() error {
+	// Update network policy to airgap
+	cmd := exec.Command("replicated", "network", "update", "policy",
+		"--id", c.network.ID,
+		"--policy=airgap")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to update network policy: %v: %s", err, string(output))
+	}
+
+	// Wait until the nodes are airgapped
+	for node := range c.Nodes {
+		if err := c.waitUntilAirgapped(node); err != nil {
+			return fmt.Errorf("failed to wait until node %d is airgapped: %v", node, err)
+		}
+	}
+
+	return nil
+}
+
+func (c *Cluster) RefreshSSHEndpoints() error {
```

Comment:
this is unused

Reply:
it was used in one of the tests that i decided to retract to merge this pr. will definitely be needed for tests that include a reset and then re-install. i agree it's unused in this pr though yeah.